### PR TITLE
Migrating to Netty 4.1

### LINKED ISSUE #164: Switch to Netty 4.1 branch
Since HTTP/2 support has been back-ported to the 4.1 branch, we should start basing our work on it since it will get us closer to an official release of Netty.

### PR DESCRIPTION
**Pull request by nmittler**
Fixes #164

### CODE REVIEW

**Comment by nmittler** [comment]
@ejona86 let's do this! :)

File: netty/src/main/java/io/grpc/transport/netty/Utils.java [hunk 1]
```diff
@@ -177,13 +173,8 @@ private static Http2Headers convertMetadata(Metadata headers) {
 
     @Override
     public EventLoopGroup create() {
-      return new NioEventLoopGroup(0, new ExecutorServiceFactory() {
```
Comment (ejona86) [inline netty/src/main/java/io/grpc/transport/netty/Utils.java]: Why this change?

Comment (nmittler) [inline netty/src/main/java/io/grpc/transport/netty/Utils.java]: ExecutorServiceFacotry and that particular constructor of NioEventLoopGroup only exist in 5.x

Comment (ejona86) [inline netty/src/main/java/io/grpc/transport/netty/Utils.java]: Doesn't this constructor give us basically the same functionality?

https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/nio/NioEventLoopGroup.java#L53-L59

Comment (nmittler) [inline netty/src/main/java/io/grpc/transport/netty/Utils.java]: The default is essentially num processors * 2 ... is that not sufficient here?

Comment (ejona86) [inline netty/src/main/java/io/grpc/transport/netty/Utils.java]: The "0" is selecting the default. Instead, we are naming the threads.

Comment (nmittler) [inline netty/src/main/java/io/grpc/transport/netty/Utils.java]: Done.

**Comment by nmittler** [comment]
@ejona86 PTAL

File: netty/src/main/java/io/grpc/transport/netty/Utils.java (hunk 1 above)
Comment (ejona86) [inline netty/src/main/java/io/grpc/transport/netty/Utils.java]: Oh, I see.

Using ThreadFactory constructor seems more what we want:
```java
return new NioEventLoopGroup(0,
    new ThreadFactoryBuilder().setNameFormat(name + "-%d").build());
```

That works, right?

Comment (nmittler) [inline netty/src/main/java/io/grpc/transport/netty/Utils.java]: oof .. done :)

**Comment by ejona86** [comment]
@nmittler, LGTM!

**Comment by nmittler** [comment]
Submitted as 2a42509

**Comment by buchgr** [comment]
awesome @nmittler !
